configure.ac

```diff
@@ -64,13 +64,18 @@ AM_CONDITIONAL([STRICT], [test "x$enable_strict" = xyes])
 
 # By default `--enable-mpi`, will set the "enable_mpi" variable
 AC_ARG_ENABLE([mpi], AS_HELP_STRING([--disable-mpi], [Disable mpi sync daemon]))
-
 AS_IF([test "x$enable_mpi" != xno], [
   AX_MPI([enable_mpi=yes], [enable_mpi=no])
   ])
-
 AM_CONDITIONAL([FOUND_MPI], [test "x$enable_mpi" != "xno"])
 
+AC_ARG_ENABLE([clang-parser], AS_HELP_STRING([--enable-clang-parser], [Enable clang parser]))
+AM_CONDITIONAL([ENABLE_CLANG_PARSER], [test "x$enable_clang_parser" = xyes])
+if test "x$enable_clang_parser" = xyes && test -z "$H2YAML"; then
+  AC_CHECK_PROG(H2YAML, [h2yaml],[ h2yaml], [no])
+  test "$H2YAML" == "no" && AC_MSG_ERROR([Required program 'h2yaml' not found.])
```

Comment:
```suggestion
  test "x$H2YAML" = "xno" && AC_MSG_ERROR([Required program 'h2yaml' not found.])
```
No? Cf above.

Reply:
The other  tests do
```
  test "$LTTNG_GEN_TP" == "no" && AC_MSG_ERROR([Required program 'lttng-gen-tp' not found.])
```
 So I don't know...

Reply:
the problem is when, for whatever reason, the variable is empty, then you test:
`test "" == "no"` which is malformed ast it is interpreted as `test == no`. If I recall correctly

Reply:
Maybe we need to fix the others as well.

Reply:
Agree, it was my recollection too. 

We also have the `=` versus `==`; I think one is `bash-ish`, and the other is the `/sh` standard...  Do you mind if we do that in another PR?

Reply:
Sure, no objection.